pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -335,6 +374,45 @@ func (ev *Evaluator) SelectCandidate(ctx context.Context, candidates []Candidate
 	return candidates[0]
 }
 
+func (ev *Evaluator) preemptPod(ctx context.Context, c Candidate, preemptor, victim *v1.Pod, pluginName string) error {
+	logger := klog.FromContext(ctx)
+
+	// If the victim is a WaitingPod, send a reject message to the PermitPlugin.
+	// Otherwise we should delete the victim.
+	if waitingPod := ev.Handler.GetWaitingPod(victim.UID); waitingPod != nil {
+		waitingPod.Reject(pluginName, "preempted")
+		logger.V(2).Info("Preemptor pod rejected a waiting pod", "preemptor", klog.KObj(preemptor), "waitingPod", klog.KObj(victim), "node", c.Name())
+	} else {
+		condition := &v1.PodCondition{
+			Type:    v1.DisruptionTarget,
+			Status:  v1.ConditionTrue,
+			Reason:  v1.PodReasonPreemptionByScheduler,
+			Message: fmt.Sprintf("%s: preempting to accommodate a higher priority pod", preemptor.Spec.SchedulerName),
+		}
+		newStatus := victim.Status.DeepCopy()
+		updated := apipod.UpdatePodCondition(newStatus, condition)
+		if updated {
+			if err := util.PatchPodStatus(ctx, ev.Handler.ClientSet(), victim, newStatus); err != nil {
+				logger.Error(err, "Could not add DisruptionTarget condition due to preemption", "pod", klog.KObj(victim), "preemptor", klog.KObj(preemptor))
+				return err
+			}
+		}
+		if err := util.DeletePod(ctx, ev.Handler.ClientSet(), victim); err != nil {
+			if !apierrors.IsNotFound(err) {
+				logger.Error(err, "Preempted pod", "pod", klog.KObj(victim), "preemptor", klog.KObj(preemptor))
+				return err
+			}
+			logger.V(2).Info("Victim Pod is already deleted", "preemptor", klog.KObj(preemptor), "victim", klog.KObj(victim), "node", c.Name())
+			return nil
```

Comment:
Maybe we don't need to return nil when pod has been deleted. Just delete this line?

Reply:
But we shouldnt reach L408 and L411, right? Those log and event would be inappropriate in this case